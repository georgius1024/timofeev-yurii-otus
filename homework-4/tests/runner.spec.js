const runner = require('../src/runner')
const path = require('path')
describe('index script', async () => {
  let oldLogProc, oldErrorProc, newLogProc, newErrorProc

  beforeEach(() => {
    oldLogProc = console.log
    oldErrorProc = console.error
    newLogProc = jest.fn()
    newErrorProc = jest.fn()
    console.log = newLogProc
    console.error = newErrorProc
  })

  it('works when valid params is specified', async () => {
    await runner(['path=./tests/foo'])
    expect(newLogProc.mock.calls).toMatchSnapshot()
    expect(newErrorProc.mock.calls.length).toBe(0)
  })

  it('helps when no params is specified', async () => {
    await runner()
    expect(newLogProc.mock.calls).toMatchSnapshot()
    expect(newErrorProc.mock.calls.length).toBe(0)
  })

  it('fails when wrong params is specified', async () => {
    await runner(['path=error'])
    expect(newLogProc.mock.calls).toMatchSnapshot()
    const errors = newErrorProc.mock.calls.map(e => e[0].code)
    expect(errors).toEqual(['ENOENT'])
  })

  afterEach(() => {
    console.log = oldLogProc
    console.error = oldErrorProc
  })
})
