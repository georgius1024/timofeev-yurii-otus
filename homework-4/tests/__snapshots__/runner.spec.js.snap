// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index script fails when wrong params is specified 1`] = `
Array [
  Array [
    "Listing for error",
  ],
]
`;

exports[`index script helps when no params is specified 1`] = `
Array [
  Array [
    "Recursive directory listing",
  ],
  Array [
    "Usage: npm run test -- path=<PATH>",
  ],
]
`;

exports[`index script works when valid params is specified 1`] = `
Array [
  Array [
    "Listing for ./tests/foo",
  ],
  Array [
    Object {
      "dirs": Array [
        "tests/foo",
        "tests/foo/bar",
        "tests/foo/bar/baz",
      ],
      "files": Array [
        "tests/foo/f1.txt",
        "tests/foo/f2.txt",
        "tests/foo/bar/bar1.txt",
        "tests/foo/bar/bar2.txt",
        "tests/foo/bar/baz/.gitignore",
      ],
    },
  ],
]
`;
